packages/@aws-cdk/aws-cloudfront-origins/lib/s3-origin.ts

```diff
@@ -11,14 +11,7 @@ import { Construct } from '@aws-cdk/core';
 /**
  * Properties to use to customize an S3 Origin.
  */
-export interface S3OriginProps {
-  /**
-   * An optional path that CloudFront appends to the origin domain name when CloudFront requests content from the origin.
-   * Must begin, but not end, with '/' (e.g., '/production/images').
-   *
-   * @default '/'
-   */
-  readonly originPath?: string;
+export interface S3OriginProps extends cloudfront.OriginProps {
```

Comment:
@njlynch Because I've decided to extend `cloudfront.OriginiProps` here now we're able to override other params other than just `originPath`, `originAccessIdentity`. It seems we can hit multiple birds with one stone as we can now pass `connectionAttempts`, `connectionTimeout`, `originShieldRegion` if desired... Do you see any issues with this? `HttpOrigin` is doing the same thing btw. Let me know what you think.

before:
![image](https://user-images.githubusercontent.com/31543/130339368-2a65cf9f-a7a6-4493-ae0c-ff1fa855908c.png)

after:
![image](https://user-images.githubusercontent.com/31543/130339481-e52c25e0-2ce6-4935-89b1-185ec9e71bb9.png)

Reply:
I think it's fine. It's certainly supported by CloudFormation/CloudFront, so we probably should take this change.

That being said, I explicitly _didn't_ include it originally because I couldn't figure out _why_ someone would want to do this. Can you help enlighten me on what type(s) of headers someone might send to an S3 origin, and for what purpose?

Reply:
Answered in #16160